File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    
    map<string, int> m;
    
    for(int i = 0 ; i < name.size() ; i++)
        m[name[i]] = yearning[i];
    
    for(auto p : photo){
        int total = 0;
        for(auto s : p)
            total += m[s];
        answer.push_back(total);
    }
    
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.stream.*;
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        Map<String, Integer> map = new HashMap<>();
        for(int i = 0 ; i < name.length ; i++)
            map.put(name[i], yearning[i]);
        
        
        return Arrays.stream(photo)
            .map(s -> (int) Arrays.stream(s)
                .map(j -> map.containsKey(j) ? map.get(j) : 0)
                .mapToInt(Integer::valueOf)
                .sum()
                )
            .mapToInt(Integer::valueOf)
            .toArray();
        
    }
}